### PR TITLE
APPS-126 -- added default environment configuration for redis setup

### PR DESCRIPTION
**Pull request by shiva-sc**

### CODE REVIEW

**Comment by shiva-sc** [comment]
@duboisp : Consider this initial draft, we can discuss more about this on Friday

File: controllers/subscriptions.js [hunk 1]
```diff
@@ -33,11 +33,25 @@ const processEnv = process.env,
...
+const redisSentinel1Uri = process.env.REDIS_SENTINEL_1_URI || '10.43.72.126';
+const redisSentinel1Port = process.env.REDIS_SENTINEL_1_PORT || '26379';
+const redisSentinel2Uri = process.env.REDIS_SENTINEL_2_URI || '10.43.72.121';
+const redisSentinel2Port = process.env.REDIS_SENTINEL_2_PORT || '26379';
+const redisMasterName = process.env.REDIS_MASTER_NAME || 'x-notify-master';
+const alertErrorTimeframe = process.env.ALERT_ERROR_TIMEFRAME || 900;
+const alertErrorThreshold = process.env.ALERT_ERROR_THRESHOLD || 3;
+
+
 const notifyQueue = new Queue('sendMail',
```
Comment (duboisp) [inline controllers/subscriptions.js:50]: Can you add logic like this code:
```javascript
if (process.env.NODE_ENV === 'production') {
    var redisConf = {
        redis: {
            sentinels: [
                { host: redisSentinel1Uri, port: redisSentinel1Port },
                { host: redisSentinel2Uri, port: redisSentinel2Port },
            ],
            name: redisMasterName,
        }
    }
} else {
    var redisConf = {
        redis: {
            host: redisUri,
            port: redisPort,
        }
    }
}
```

**Review by duboisp — CHANGES_REQUESTED** [review]

**Review by duboisp — COMMENTED** [review]
Don't forget to document those new environment variable here: https://github.com/ServiceCanada/x-notify/blob/master/setup.md#environment-variables

File: notifyQueue.js [hunk 2]
```diff
@@ -2,11 +2,37 @@ const Queue = require('bull');
...
+const redisPort = process.env.REDIS_PORT || '6379';
+const redisSentinel1Uri = process.env.REDIS_SENTINEL_1_URI || '127.0.0.1';
+const redisSentinel1Port = process.env.REDIS_SENTINEL_1_PORT || '26379';
+const redisSentinel2Uri = process.env.REDIS_SENTINEL_2_URI || '127.0.0.1';
+const redisSentinel2Port = process.env.REDIS_SENTINEL_2_PORT || '26379';
+const redisMasterName = process.env.REDIS_MASTER_NAME || 'x-notify-master';
+const alertErrorTimeframe = process.env.ALERT_ERROR_TIMEFRAME || 900;
+const alertErrorThreshold = process.env.ALERT_ERROR_THRESHOLD || 3;
+
 const notifyQueue = new Queue('sendMail',
```
Comment (duboisp) [inline notifyQueue.js:17]:
```suggestion

if ( .... ){ 
redisConf = {}
else {
redisConf = {}
}

const notifyQueue = new Queue('sendMail', redisConf );
```

**Review by duboisp — CHANGES_REQUESTED** [review]

**Review by duboisp — CHANGES_REQUESTED** [review]
To not forget, this PR are only missing the documentation of the environment variable and the associated JIRA would need to include special instructions for when the application would be deployed.

**Comment by shiva-sc** [comment]
@duboisp : I have modified the documentation for environment variable. Can you please review?

File: setup.md [hunk 3]
```diff
@@ -59,70 +59,81 @@ Server will run at `0.0.0.0:8080` by default.
...
-`jwtSecretAllSubsForTopic` This is the secret key necessary for authentication via JWT to the confirmedSubscribers service
+* `subsLinkSuffix` Suffix buffer URL parameter for when we send confirm/remove link. Default: "853e0212b92a127"
+* `jwtSecretAllSubsForTopic` This is the secret key necessary for authentication via JWT to the confirmedSubscribers service
 
-`baseFolder` Base folder where the application run. ex: "/x-notify" Default: undefined
+* `baseFolder` Base folder where the application run. ex: "/x-notify" Default: undefined
+
+
+### REDIS Default Configuration
+* `REDIS_ENV` Set environment value for Redis. Default: `stage`
```
Comment (duboisp) [inline setup.md:129]:
```suggestion
* `REDIS_ENV` Set environment value for Redis. Default: `stage` and for production use which leverage the redis-sentinel, this need to be set to `production`
```

**Review by duboisp — CHANGES_REQUESTED** [review]
Other that the small clarification, the rest look good.

We would only need to provide some instruction in our jira request for the deployment on stage and for the deployment on prod, like what env variable that need to be defined.

**Review by duboisp — APPROVED** [review]

File: notifyQueue.js [hunk 4]
```diff
@@ -2,15 +2,37 @@ const Queue = require('bull');
...
+const redisPort = process.env.REDIS_PORT || '6379';
+const redisSentinel1Uri = process.env.REDIS_SENTINEL_1_URI || '127.0.0.1';
+const redisSentinel1Port = process.env.REDIS_SENTINEL_1_PORT || '26379';
+const redisSentinel2Uri = process.env.REDIS_SENTINEL_2_URI || '127.0.0.1';
+const redisSentinel2Port = process.env.REDIS_SENTINEL_2_PORT || '26379';
+const redisMasterName = process.env.REDIS_MASTER_NAME || 'x-notify-master';
+
+
+let redisConf = {};
+if (process.env.NODE_ENV === 'production') {
```
Comment (duboisp) [inline notifyQueue.js:16]: According to documentation (line 77 in setup.md), the value need to be set to:  `prod`

**Review by duboisp — CHANGES_REQUESTED** [review]

**Review by duboisp — APPROVED** [review]

**Pull request merged**